packages/contracts/contracts/MVM/MVM_InboxSenderManager.sol

```diff
@@ -0,0 +1,111 @@
+    {
+        for (uint256 i = 0; i < _defaultInboxSenders.length; ++i) {
+            defaultInboxSender[_defaultInboxSenders[i].senderType] = _defaultInboxSenders[i].sender;
+        }
+    }
+
+    /**********************
+     * Function Modifiers *
+     **********************/
+    modifier onlyManager() {
+        require(
+            msg.sender == resolve("METIS_MANAGER"),
+            "MVM_InboxSenderManager: Function can only be called by the METIS_MANAGER."
+        );
+        _;
+    }
+
+    /********************
+     * Public Functions *
+     ********************/
+    function setInboxSenders(uint256 blockNumber, InboxSender[] calldata _inboxSenders)
+        external
+        override
+        onlyManager
+    {
+        _setInboxSenders(blockNumber, _inboxSenders);
+    }
+
+    // allow us to overwrite the last block number and its senders, just in case if we made any mistake
+    function overwriteLastInboxSenders(uint256 blockNumber, InboxSender[] calldata _inboxSenders)
+        external
+        override
+        onlyManager
+    {
+        require(blockNumbers.length > 0, "MVM_InboxSenderManager: No block to update.");
+
+        // pop the last block
+        uint256 lastBlockNumber = blockNumbers[blockNumbers.length - 1];
+        blockNumbers.pop();
+
+        // clean up the last block senders
+        mapping(InboxSenderType => InboxSender) storage lastInboxSenders = inboxSenders[lastBlockNumber];
+        delete lastInboxSenders[InboxSenderType.InboxSender];
+        delete lastInboxSenders[InboxSenderType.InboxBlobSender];
+
+        // write the new senders
+        _setInboxSenders(blockNumber, _inboxSenders);
+    }
+
+    function getInboxSender(uint256 blockNumber, InboxSenderType inboxSenderType) external view override returns (address) {
+        for (int256 i = int256(blockNumbers.length) - 1; i >= 0; i--) {
+            if (blockNumbers[uint256(i)] <= blockNumber) {
+                address sender = inboxSenders[blockNumbers[uint256(i)]][inboxSenderType].sender;
+                if (sender != address(0)) {
+                    return sender;
+                }
+            }
+        }
+
+        return defaultInboxSender[inboxSenderType];
+    }
+
+    /********************
+     * Internal Functions *
+     ********************/
+    function _setInboxSenders(uint256 blockNumber, InboxSender[] calldata _inboxSenders)
+    private
+    {
+        if (blockNumbers.length > 0) {
+            require(
+                blockNumber > blockNumbers[blockNumbers.length - 1],
+                "MVM_InboxSenderManager: Block number must be greater than the previous block number."
+            );
+        }
+        for (uint256 i = 0; i < _inboxSenders.length; ++i) {
+            inboxSenders[blockNumber][_inboxSenders[i].senderType] = _inboxSenders[i];
+        }
+        blockNumbers.push(blockNumber);
+        for (uint256 i = 0; i < _inboxSenders.length; ++i) {
```

Comment:
You don't need to have two loops for it.

Reply:
Fixed